### PR TITLE
[NetAppFiles]Add 2021-10-01 changes

### PR DESCRIPTION
**Pull request by audunn**
## Description

This updates the NetApp Files module to support features in api-version 2021-10-01

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1172
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by audunn** [comment]
note parameter name `Cifs` that static analysis is complaining about  is the name of a network filesystem protocol not the plural form of anything.

File: src/NetAppFiles/NetAppFiles/Snapshot/RestoreFilesNetAppFilesSnapshot.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+using Microsoft.Azure.Commands.NetAppFiles.Helpers;
+using Microsoft.Azure.Commands.NetAppFiles.Models;
+using Microsoft.Azure.Management.NetApp;
+using System.Linq;
+using Microsoft.Azure.Management.NetApp.Models;
+
+namespace Microsoft.Azure.Commands.NetAppFiles.Snapshot
+{
+    [Cmdlet(
+        "RestoreFiles",
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/Snapshot/RestoreFilesNetAppFilesSnapshot.cs:28]: Hi @audunn, is this command expected to add? I didn't saw its help file. BTW, "RestoreFiles" is not approved verb in powershell.

File: src/NetAppFiles/NetAppFiles/Subvolume/RemoveNetAppFilesSubvolume.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            }
+
+            if (ShouldProcess(Name, string.Format(PowerShell.Cmdlets.NetAppFiles.Properties.Resources.RemoveResourceMessage, Name)))
+            {
+                AzureNetAppFilesManagementClient.Subvolumes.Delete( ResourceGroupName, accountName: AccountName, poolName: PoolName, volumeName: VolumeName, subvolumeName: Name);                
+            }
+
+            if (PassThru)
+            {
+                WriteObject(success);
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/Subvolume/RemoveNetAppFilesSubvolume.cs:158]: How can we let success be true? In common practice, if this command runs successfully, we will return true.

File: src/NetAppFiles/NetAppFiles/Volume/NewNetAppFilesVolume.cs [hunk 3]
```diff
@@ -220,7 +221,9 @@ public class NewAzureRmNetAppFilesVolume : AzureNetAppFilesCmdletBase
         [Parameter(
             Mandatory = false,
             HelpMessage = "UNIX permissions for NFS volume accepted in octal 4 digit format. First digit selects the set user ID(4), set group ID (2) and sticky (1) attributes. Second digit selects permission for the owner of the file: read (4), write (2) and execute (1). Third selects permissions for other users in the same group. the fourth for other users not in the group. 0755 - gives read/write/execute permissions to owner and read/execute to group and other users.")]
-        public string UnixPermissions { get; set; }
+        [CmdletParameterBreakingChange("UnixPermissions", ChangeDescription = "Parameter Alias UnixPermissions will be removed, please use  UnixPermission.")]
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/Volume/NewNetAppFilesVolume.cs:224]:
```suggestion
        [CmdletParameterBreakingChange("UnixPermissions", ChangeDescription = "Parameter Alias UnixPermissions will be removed, please use UnixPermission.")]
```

File: src/NetAppFiles/NetAppFiles/VolumeGroup/NewNetAppFilesVolumeGroup.cs [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+                volumeExportPolicy = ModelExtensions.ConvertExportPolicyFromPs(ExportPolicy);
+            }
+
+            return CreateHostVolumeGroup(name, ApplicationIdentifier, poolResourceId, tagPairs, BackupProtocolType, volumeExportPolicy, StartingHostId, HostCount);
+            
+        }
+
+        private VolumeGroupDetails CreateHostVolumeGroup(string name, string sid, string poolResourceId, IDictionary<string, string> tagPairs, string[] volumeBackupProtocolTypes, VolumePropertiesExportPolicy volumeExportPolicy, int startingHostId,  int hostCount)
+        {
+            var globalPlacementRules = new List<PlacementKeyValuePairs> { new PlacementKeyValuePairs { Key = "Key1", Value = "value1" } };
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/VolumeGroup/NewNetAppFilesVolumeGroup.cs:391]: Why here is a hard-code GlobalPlacementRule. I saw this parameter is hiden from cmdlet, is it expected?

File: src/NetAppFiles/NetAppFiles/help/Az.NetAppFiles.md [hunk 5]
```diff
@@ -41,6 +41,12 @@ Gets details of an Azure NetApp Files (ANF) snapshot.
 ### [Get-AzNetAppFilesSnapshotPolicy](Get-AzNetAppFilesSnapshotPolicy.md)
 Gets details of an Azure NetApp Files (ANF) snapshot policy.
 
+### [Get-AzNetAppFilesSubvolume](Get-AzNetAppFilesSubvolume.md)
+Gets details of an Azure NetApp Files (ANF) volume.
+
+### [Get-AzNetAppFilesSubvolumeMetadata](Get-AzNetAppFilesSubvolumeMetadata.md)
+{{ Fill in the Synopsis }}
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Az.NetAppFiles.md:48]: Please fill this placeholder.

File: src/NetAppFiles/NetAppFiles/help/Az.NetAppFiles.md [hunk 6]
```diff
@@ -50,12 +56,21 @@ Gets details of an Azure NetApp Files (ANF) volume.
...
 Get volume's restore status
 
 ### [Initialize-AzNetAppFilesReplication](Initialize-AzNetAppFilesReplication.md)
 Re-Initializes the replication connection on the destination volume
 
+### [New-AzExportPolicyObject](New-AzExportPolicyObject.md)
+Creates export policy object.
+
+### [New-AzExportPolicyRuleObject](New-AzExportPolicyRuleObject.md)
+{{ Fill in the Synopsis }}
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Az.NetAppFiles.md:72]: To be finished.

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolume.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+## DESCRIPTION
+The **Get-AzNetAppFilesVolume** cmdlet gets details of an ANF subvolume.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> Get-AzNetAppFilesVolume -ResourceGroupName "MyRG" -AccountName "MyAnfAccount" -PoolName "MyAnfPool" -VolumeName "MyAnfVolume" -Name "MyAnfSubvolume"
+```
+
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolume.md:42]: Could you provide sample output?

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolume.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+### Microsoft.Azure.Commands.NetAppFiles.Models.PSNetAppFilesVolume
+
+## OUTPUTS
+
+### Microsoft.Azure.Commands.NetAppFiles.Models.PSNetAppFilesVolume
+
+## NOTES
+
+## RELATED LINKS
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolume.md:182]: Please add the reference link of related cmdlet here.

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolumeMetadata.md [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+external help file: Microsoft.Azure.PowerShell.Cmdlets.NetAppFiles.dll-Help.xml
+Module Name: Az.NetAppFiles
+online version:
+schema: 2.0.0
+---
+
+# Get-AzNetAppFilesSubvolumeMetadata
+
+## SYNOPSIS
+{{ Fill in the Synopsis }}
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolumeMetadata.md:11]: Please fill this placeholder.

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolumeMetadata.md [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+```
+
+### ByParentObjectParameterSet
+```
+Get-AzNetAppFilesSubvolumeMetadata -Name <String> -VolumeObject <PSNetAppFilesVolume>
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+## DESCRIPTION
+{{ Fill in the Description }}
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolumeMetadata.md:34]: There are many placeholder to be finished.

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolumeMetadata.md [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> {{ Add example code here }}
+```
+
+{{ Add example description here }}
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesSubvolumeMetadata.md:43]: Please provide example here

**Review by BethanyZhou — COMMENTED** [review]
Please check whether there are some place holder in help MDs and fill them.

File: src/NetAppFiles/NetAppFiles/Snapshot/RestoreFilesNetAppFilesSnapshot.cs (hunk 1 above)
Comment (BethanyZhou) [reply src/NetAppFiles/NetAppFiles/Snapshot/RestoreFilesNetAppFilesSnapshot.cs]: Sounds good to me.

File: src/NetAppFiles/NetAppFiles/help/Restore-AzNetAppFilesSnapshotFile.md [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+external help file: Microsoft.Azure.PowerShell.Cmdlets.NetAppFiles.dll-Help.xml
+Module Name: Az.NetAppFiles
+online version:
+schema: 2.0.0
+---
+
+# Restore-AzNetAppFilesSnapshotFile
+
+## SYNOPSIS
+{{ Fill in the Synopsis }}
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Restore-AzNetAppFilesSnapshotFile.md:11]: Please fill placeholder

**Review by BethanyZhou — COMMENTED** [review]
Looks good to me. Besides, please check whether there is any reference link can be added in RELATED LINKS section for each command.

File: src/NetAppFiles/NetAppFiles/help/Restore-AzNetAppFilesSnapshotFile.md (hunk 12 above)
Comment (audunn) [reply src/NetAppFiles/NetAppFiles/help/Restore-AzNetAppFilesSnapshotFile.md:11]: hmm I did, must have gotten lost after regeneration of docs, will add.

Comment (audunn) [reply src/NetAppFiles/NetAppFiles/help/Restore-AzNetAppFilesSnapshotFile.md:11]: btw there are a lot of RELATED LINKS that could be added, do we have a guidance for this section?

**Comment by BethanyZhou** [comment]
> New

Yes, that would be better to name cmdlet with pattern "Verb-Az{ModuleName}{Noun}"

File: src/NetAppFiles/NetAppFiles/help/Restore-AzNetAppFilesSnapshotFile.md (hunk 12 above)
Comment (BethanyZhou) [reply src/NetAppFiles/NetAppFiles/help/Restore-AzNetAppFilesSnapshotFile.md]: No guidance. they are nice to have.

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesVolumeGroup.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
+
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesVolumeGroup.md:1]:
```suggestion
---
```

File: src/NetAppFiles/NetAppFiles/Snapshot/RestoreNetAppFilesSnapshotFiles.cs [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+                SnapshotRestoreFiles snapshotRestoreFilesBody = new SnapshotRestoreFiles()
+                {
+                    FilePaths = FilePath,
+                    DestinationPath = DestinationPath
+                };
+
+                AzureNetAppFilesManagementClient.Snapshots.RestoreFiles(snapshotRestoreFilesBody, ResourceGroupName, AccountName, PoolName, VolumeName, Name);
+                success = true;
+            }
+            if (PassThru)
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/Snapshot/RestoreNetAppFilesSnapshotFiles.cs:151]:
```suggestion
            if (PassThru.IsPresent)
```
Not check the syntax whether successes

File: src/NetAppFiles/NetAppFiles/Snapshot/RestoreNetAppFilesSnapshotFiles.cs [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+                AccountName = parentResources[1];
+                PoolName = parentResources[3];
+                VolumeName = parentResources[5];
+                Name = resourceIdentifier.ResourceName;
+            }
+            else if (ParameterSetName == ParentObjectParameterSet)
+            {
+                ResourceGroupName = VolumeObject.ResourceGroupName;
+                var NameParts = VolumeObject.Name.Split('/');
+                AccountName = NameParts[0];
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/Snapshot/RestoreNetAppFilesSnapshotFiles.cs:135]: Maybe we should check the length of NameParts whether is greater than 2

File: src/NetAppFiles/NetAppFiles/help/Approve-AzNetAppFilesReplication.md [hunk 16]
```diff
@@ -231,3 +231,13 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 ## NOTES
 
 ## RELATED LINKS
+
+[Inititialize-AzNetAppFilesReplication](./Approve-AzNetAppFilesReplication.md)
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Approve-AzNetAppFilesReplication.md:235]: Why the link of Inititialize-AzNetAppFilesReplication points to Approve-AzNetAppFilesReplication.md

**Comment by audunn** [comment]
@BethanyZhou PR updated

**Comment by audunn** [comment]
@BethanyZhou  @wyunchi-ms   can we move this along now?

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesVault.md [hunk 17]
```diff
@@ -134,3 +134,15 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
...
+[Remove-AzNetAppFilesBackup](./Remove-AzNetAppFilesBackup.md)
+[Update-AzNetAppFilesBackup](./Update-AzNetAppFilesBackup.md)
+[Get-AzNetAppFilesBackupPolicy](./Get-AzNetAppFilesBackupPolicy.md)
+[New-AzNetAppFilesBackupPolicy](./New-AzNetAppFilesBackupPolicy.md)
+[Update-AzNetAppFilesBackupPolicy](./Update-AzNetAppFilesBackupPolicy.md)
+[Remove-AzNetAppFilesBackupPolicy](./Remove-AzNetAppFilesBackupPolicy.md)
+[Get-AzNetAppFilesVolume](./Get-AzNetAppFilesAzNetAppFilesVolume.md)
+[New-AzNetAppFilesVolume](./New-AzNetAppFilesAzNetAppFilesVolume.md)
+[Update-AzNetAppFilesVolume](./Update-AzNetAppFilesAzNetAppFilesVolume.md)
+[Remove-AzNetAppFilesVolume](./Remove-AzNetAppFilesAzNetAppFilesVolume.md)
```
Comment (BethanyZhou) [inline src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesVault.md:148]:
```suggestion
[Get-AzNetAppFilesVolume](./Get-AzNetAppFilesVolume.md)
[New-AzNetAppFilesVolume](./New-AzNetAppFilesVolume.md)
[Update-AzNetAppFilesVolume](./Update-AzNetAppFilesVolume.md)
[Remove-AzNetAppFilesVolume](./Remove-AzNetAppFilesVolume.md)
```
Please check all links in other markdowns.

**Review by BethanyZhou — COMMENTED** [review]
Generally looks good to me except one comment.

File: src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesVault.md (hunk 17 above)
Comment (audunn) [reply src/NetAppFiles/NetAppFiles/help/Get-AzNetAppFilesVault.md:148]: good catch, will fix thanks

**Review by BethanyZhou — APPROVED** [review]
Looks good to me

**Pull request merged**